### PR TITLE
[sha3] Make error report consistent

### PR DESCRIPTION
**Pull request by eunchan**
In previous design, SHA3 may or may not report the error for double
process commands depending on the current SHA3 state. If SHA3 state sits
in `StAbsorb` state, double `process` commands were accepted and
triggered `keccak_process` twice. (It does not harm the sha3 hashing
operation.)
    
This commit introduces `processing` status bit to check more than one
`process` commands while the state is in `StAbsorb`. If additional
process command is received, regardless of the state, the SHA3 reports
an error with the error code `ErrSha3SwControl`.

### CODE REVIEW

File: hw/ip/kmac/rtl/kmac.sv
Comment (udinator) [inline hw/ip/kmac/rtl/kmac.sv:447]: is this prioritization right?

if i start an App transaction and decide to write CmdProcess during the App transaction, the errchecker will also report a sequencing error, but this one will be prioritized over the `ErrSwIssuedCmdInAppActive` from `kmac_app` - seems like the kmac_app error should be the first one reported in this scenario.

Comment (eunchan) [reply hw/ip/kmac/rtl/kmac.sv:447]: Hum.. make sense. Let me revise the original PR.

File: hw/ip/kmac/doc/_index.md
Comment (mwbranstad) [inline hw/ip/kmac/doc/_index.md:309]: How big is this expiration timer? EDN may have to wait up to 5000 usec for an entropy ack in the worst case.

File: hw/ip/kmac/rtl/sha3.sv
Comment (mwbranstad) [inline hw/ip/kmac/rtl/sha3.sv:111]: Any new port changes to sha3.sv?

**Review by mwbranstad — APPROVED** [review]
LGTM, left a couple of questions.

File: hw/ip/kmac/doc/_index.md
Comment (eunchan) [reply hw/ip/kmac/doc/_index.md:309]: Um.. 16 bits timer currently 16 bit (65535) , and KMAC may run ~ 100MHz, So,, definitely not enough!... Thanks for notifying that.. I should have some prescaler there.

**Review by udinator — APPROVED** [review]

**Comment by eunchan** [comment]
@imphil Looks like the CI fails at the CW310 test always. Sometimes, negative slack, which I cannot reproduce in my local, other times tooling issue (install failure on OpenOCD 0.11). Could you let me know what could be the next step to resolve this?

**Comment by imphil** [comment]
Sorry for having bad luck with CI.

* The "cannot install openocd" issue is infrastructure-related (a package mirror not under our control is sometimes serving outdated information) and has nothing to do with this PR. It's also resolved (for) now.
* The negative slack is actually something that has come up before and we weren't able to figure out where it exactly came from. It's a small hold time violation. https://github.com/lowRISC/opentitan/issues/6772 was the previous issue on that, and @GregAC had a look and couldn't reproduce it or find something obvious in the timing reports. So maybe retriggering makes it go away for you as well, but I suspect this topic will come up again (it's apparently low-probability enough not to be noticed too often). So if you can reproduce it locally and have a look at the timing reports that would be great.

**Comment by eunchan** [comment]
> Sorry for having bad luck with CI.
> 
> * The "cannot install openocd" issue is infrastructure-related (a package mirror not under our control is sometimes serving outdated information) and has nothing to do with this PR. It's also resolved (for) now.
> * The negative slack is actually something that has come up before and we weren't able to figure out where it exactly came from. It's a small hold time violation. #6772 was the previous issue on that, and @GregAC had a look and couldn't reproduce it or find something obvious in the timing reports. So maybe retriggering makes it go away for you as well, but I suspect this topic will come up again (it's apparently low-probability enough not to be noticed too often). So if you can reproduce it locally and have a look at the timing reports that would be great.

Thanks. I tried multiple times, but failed (almost 10 times?) consistently. Anyway, let me merge the PR if the same issue occurs.

**Comment by rswarbrick** [comment]
Please do not merge this PR if it will break CI on master.

**Comment by eunchan** [comment]
> Please do not merge this PR if it will break CI on master.
> 
> When I was investigating #6772, it was deterministic. We probably need to fix the timing problem as an urgent matter, but that doesn't mean we should break the master branch.

OK. Understand. Is there a way to download all collaterals from CI. I want to get the timing report for the failure but the PIpelines log does not show the path.

**Comment by imphil** [comment]
> I want to get the timing report for the failure but the PIpelines log does not show the path.

That's currently not generated/uploaded. I did dig out an older commit of mine to upload the whole build directory and pushed it at https://github.com/lowRISC/opentitan/pull/7017 now. If you need additional reports from Vivado, you can add them to the necessary TCL commands to hw/top_earlgrey/util/vivado_hook_write_bitstream_pre.tcl for example.

**Comment by eunchan** [comment]
OK passed now (somehow?) Let me merge this PR.

**Pull request merged**
